File: Длинный тур отборочного этапа Открытой олимпиады школьников 2023_2024 учебного года/checker/gen.py
from random import *

n = randrange(1, 3)
print(n)
n = 2 ** n
a = [randrange(1, 10) for i in range(n)]
print(*a)
m = randrange(1, 4)
print(m)
for i in range(m):
    t = randrange(1, 3)
    v = -1
    l, r, k = randrange(0, n), randrange(0, n), randrange(0, n)
    if (t == 1):
        v = randrange(1, 10)
        print(t, l, r, k, v)
    else:
        print(t, l, r, k)
    

File: 101223/checker/gen.cpp
#include <iostream>
#include <vector>
#include <chrono>
#include <random>
using namespace std;

ostream& operator<< (ostream& stream, const vector<int>& a) {
    for (auto to : a) {
        stream << to << " ";
    }
    return stream;
}

auto now = chrono::high_resolution_clock::now().time_since_epoch().count();
mt19937_64 rnd(now);

long long randrange(long long a, long long b) {
    return a + rnd() % (b - a);
}

int main() {
    for (int i = 0; i < 8; i++) {
        cout << randrange(-100, 500) << " ";
    }
}
